File: Python/Binary to denary conversion.py
bits_list = []
user_input = input("Enter binary code (1 Byte): ")

for bit in user_input:
    bits_list.append(int(bit))

weighting_list = [128, 64, 32, 16, 8, 4, 2, 1]
weighting_index = 0
total = 0

for bit in bits_list:
    if bit == 1:
        total = total + weighting_list[weighting_index]
    weighting_index = weighting_index + 1
    
print(f"The denary value is: {total}")

File: Javascript/Binary to denary conversion.js
let bits_list = [];
let user_input = prompt("Enter binary code (1 Byte): ")

for (let index = 0; index <= user_input.length; index++) {
    bits_list.push(Number(user_input[index]))
};

let weighting_list = [128, 64, 32, 16, 8, 4, 2, 1];
let weighting_index = 0;
let total = 0;

for (let index = 0; index <= bits_list.length; index++) {
    if (bits_list[index] === 1) {
        total = total + weighting_list[index]
    };
};

alert("The denary value is: " + total)
